plugins/cybereason/icon_cybereason/actions/quarantine_file/action.py

```diff
@@ -16,106 +16,19 @@ def __init__(self):
 
     def run(self, params={}):
         sensor_details = self.connection.api.get_sensor_details(params.get(Input.SENSOR))
-        sensor_guid = sensor_details["guid"]
         malop_id = params.get(Input.MALOP_ID)
         quarantine = params.get(Input.QUARANTINE)
-
         malop_data = self.connection.api.get_malop(malop_id)
+        sensor_guid = sensor_details["guid"]
+        sensor_machine_name = sensor_details["machineName"]
 
-        if not self.check_machine_in_malop(malop_data, sensor_guid):
-            raise PluginException(
-                cause="Sensor provided is not related to the Malop ID Provided.",
-                assistance=f"Make sure the provided sensor is involved in the Malop: {malop_id}.",
-            )
-
-        file_guids = self.get_file_guids(
-            self.get_files_in_malop(self.connection.api.get_malop(malop_id)),
-            sensor_details["machineName"],
-            sensor_guid,
-            quarantine,
-        )
-
-        if quarantine:
-            action_type = "QUARANTINE_FILE"
-        else:
-            action_type = "UNQUARANTINE_FILE"
-
-        actions = []
-        for guid in file_guids:
-            actions.append({"targetId": guid, "actionType": action_type})
-
-        if not actions:
-            raise PluginException(
-                cause="No actions to perform.",
-                assistance="This can happen because there are no quarantined files in the provided Malop.",
-            )
+        CybereasonAPI.check_machine_in_malop(malop_data, sensor_guid, malop_id)
 
+        files_in_malop = CybereasonAPI.get_files_in_malop(malop_data)
+        file_guids = self.connection.api.get_file_guids(files_in_malop, sensor_machine_name, quarantine)
```

Comment:
Should this be `CybereasonAPI.get_file_guids` ?

Reply:
some of the methods in the api were labeled as static methods and seemed to behave differently. this was the only way to call them.